dbt/include/snowflake/macros/materializations/dynamic_table/ddl.sql

```diff
@@ -1,46 +1,96 @@
-{% macro snowflake__get_alter_dynamic_table_as_sql(
-    relation,
+{% macro snowflake__alter_dynamic_table_sql(
+    dynamic_table,
     configuration_changes,
-    sql,
     existing_relation,
     backup_relation,
     intermediate_relation
 ) -%}
-    {{- log('Applying ALTER to: ' ~ relation) -}}
-    {{- snowflake__get_replace_dynamic_table_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) -}}
+    {{- log('Applying ALTER to: ' ~ dynamic_table.path) -}}
+
+    {% if configuration_changes.requires_full_refresh %}
+
+        {{ snowflake__replace_dynamic_table_sql(dynamic_table, existing_relation, backup_relation, intermediate_relation) }}
+
+    -- otherwise apply individual changes as needed
+    {% else %}
+
+        {%- set target_lag = configuration_changes.target_lag -%}
+        {%- if target_lag -%}{{- log('Applying UPDATE TARGET_LAG to: ' ~ dynamic_table.path) -}}{%- endif -%}
+        {%- set warehouse = configuration_changes.warehouse -%}
+        {%- if warehouse -%}{{- log('Applying UPDATE WAREHOUSE to: ' ~ dynamic_table.path) -}}{%- endif -%}
+
+        alter dynamic table {{ dynamic_table.path }} set
+            {% if target_lag %}target_lag = '{{ target_lag.context }}'{% endif %}
+            {% if warehouse %}warehouse = {{ warehouse.context }}{% endif %}
+
+    {%- endif -%}
+
 {%- endmacro %}
 
 
-{% macro snowflake__get_create_dynamic_table_as_sql(relation, sql) -%}
-    {{- log('Applying CREATE to: ' ~ relation) -}}
+{% macro snowflake__create_dynamic_table_sql(dynamic_table) -%}
+    {{- log('Applying CREATE to: ' ~ dynamic_table.path) -}}
 
-    create or replace dynamic table {{ relation }}
-        lag = '{{ config.get("target_lag") }}'
-        warehouse = {{ config.get("warehouse") }}
-        as ({{ sql }})
+    create or replace dynamic table {{ dynamic_table.path }}
+        target_lag = '{{ dynamic_table.target_lag }}'
+        warehouse = {{ dynamic_table.warehouse }}
+        as ({{ dynamic_table.query }})
```

Comment:
Okay so I _think_ this is one of those places that without spacing to

```sql
as (
  {{ dynamic_table.query }}
)
```
would result in bugs. We've seen this in RS (or whichever db 🙈 ). Did you prescrub out comments? If not, we should probably code defensively here.

Reply:
Ah yes, I forgot about that. Nice catch. To be honest, I made it look like that because there were extra returns in the logs and this looked prettier, lol. Will change.